pkg/action/scan.go

```diff
@@ -86,6 +101,10 @@ func scanSinglePath(ctx context.Context, c malcontent.Config, path string, ruleF
 		return nil, err
 	}
 
+	if fr.Error != "" {
+		return &malcontent.FileReport{Path: path, Error: fmt.Sprintf("generate: %v", fr.Error)}, nil
+	}
+
```

Comment:
It's a little confusing that errors can be set in `err` or `fr.Error` - can we consolidate these? (e.g. should we have a custom error type that wraps the error message + path?)

Reply:
That's a good question. We do this in three other places and it throws me off since I would expect to just return `err` or `fmt.Errorf(...)`.

Reply:
I'll fix this up in another PR. We use `fr.Error` to change rendering behavior elsewhere so it's a more systemic change than swapping to a new return pattern.

Really appreciate the suggestion, though 👍🏻